bookkeeper-benchmark/src/test/java/org/apache/bookkeeper/benchmark/TestBenchmark.java

```diff
@@ -66,27 +81,20 @@ public void testBookie() throws Exception {
 
     @Test
     public void testReadThroughputLatency() throws Exception {
-        final AtomicBoolean threwException = new AtomicBoolean(false);
-        Thread t = new Thread() {
-                public void run() {
-                    try {
-                        BenchReadThroughputLatency.main(new String[] {
-                                "--zookeeper", zkUtil.getZooKeeperConnectString(),
-                                "--listen", "10"});
-                    } catch (Throwable t) {
-                        LOG.error("Error reading", t);
-                        threwException.set(true);
-                    }
-                }
-            };
-        t.start();
+        ExecutorService executor = Executors.newSingleThreadExecutor(
+                new ThreadFactoryBuilder().setNameFormat("read-throughput-latency-test").build());
+        Future<Integer> f = executor.submit(() -> {
+                BenchReadThroughputLatency.main(new String[] {
+                        "--zookeeper", zkUtil.getZooKeeperConnectString(),
+                        "--listen", "10"});
+                return 0;
+            });
 
-        Thread.sleep(10000);
+        Thread.sleep(2000);
```

Comment:
since you are changing here, can we improve it to get rid of the sleep logic here?

Reply:
Would be very hard to without changing the ReadThroughputLatency benchmark completely. That's out of scope for the fix. I can remove all changes that are unrelated to the core problem, but this fix needs to go in ASAP. Postcommit[1] has been broken for 9 days now.

[1] https://builds.apache.org/view/B/view/BookKeeper/job/bookkeeper_postcommit_master_java8/

Reply:
personally I would suggest fixing the problem only rather than coupling with additional refactor. because it seems that this refactor doesn't really address any problem, which it seems to be unnecessary for fixing the CI.

Reply:
The refactor reduces the time we spend waiting. Will remove it though.